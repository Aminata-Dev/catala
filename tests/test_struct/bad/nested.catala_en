## Article

```catala
declaration enumeration E:
  -- Empty
  -- Rec content E

declaration scope A:
  context y content E

scope A:
  definition y equals E.Empty
```

```catala-test-inline
$ catala Interpret -s A
[WARNING] In scope "A", the variable "y" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_struct/bad/nested.catala_en:9.10-11:
└─┐
9 │   context y content E
  │           ‾
  └─ Article
[WARNING] The constructor "Rec" of enumeration "E" is never used; maybe it's unnecessary?

┌─⯈ tests/test_struct/bad/nested.catala_en:6.5-8:
└─┐
6 │   -- Rec content E
  │      ‾‾‾
  └─ Article
[ERROR] The type E is defined using itself, which is forbidden since Catala does not provide recursive types

┌─⯈ tests/test_struct/bad/nested.catala_en:6.17-18:
└─┐
6 │   -- Rec content E
  │                  ‾
  └─ Article
#return code 255#
```
