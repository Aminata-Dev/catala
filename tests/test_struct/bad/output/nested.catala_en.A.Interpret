[ERROR] The type S is defined using itself, which is forbidden since Catala does not provide recursive types

 --> tests/test_struct/bad/nested.catala_en
  | 
5 |   data x content S
  |                  ^
  + Article
