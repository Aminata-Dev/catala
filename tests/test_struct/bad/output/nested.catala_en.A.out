[ERROR] The type S is defined using itself, which is forbidden since Catala does not provide recursive types

 --> test_struct/bad/nested.catala_en
  | 
5 |   data x content S
  |                  ^
  + Article
